src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java

```diff
@@ -37,13 +45,71 @@ public class LogCacheClient {
     private final OAuthClient oAuthClient;
     private final Map<String, String> requestTags;
 
+//    private final org.cloudfoundry.logcache.v1.LogCacheClient delegate;
+
     public LogCacheClient(String logCacheApi, OAuthClient oAuthClient, Map<String, String> requestTags) {
         this.logCacheApi = logCacheApi;
         this.oAuthClient = oAuthClient;
         this.requestTags = requestTags;
     }
 
-    public Flux<ApplicationLogEntity> getRecentLogs(UUID applicationGuid, LocalDateTime offset) {
+    private ApplicationLog mapToAppLog(Envelope envelope) {
```

Comment:
Sort the methods so they are chained and maybe rename to **mapEnvelopeTpAppLog**